qiskit/circuit/quantumcircuit.py

```diff
@@ -403,6 +403,8 @@ def reverse_bits(self):
         circ = QuantumCircuit(
             *reversed(self.qregs),
             *reversed(self.cregs),
+            [*reversed(self.qubits)],
+            [*reversed(self.clbits)],
```

Comment:
`self.qubits` isn't distinct from `self.qregs`; `self.qregs` contains all the quantum registers, and `self.qubits` contains all qubits, _including_ those that are also contained in registers.  You might want to look at how the function `QuantumCircuit.qasm` determines whether a qubit/clbit has no register.

Reply:
Thanks for pointing this out @jakelishman . @apcarrik I think we would be able to overcome this error if you could first load the reversed `self.qubits` (something like `list(reversed(self.qubits))`) and reversed `self.clbits` in the circuits first then followed by reversed registers (something like `*reversed(self.qregs)`). This will help because when a register is added to a circuit, only those qubits that are in the register but not in `self.qubits` gets added to `self.qubits`. This ensures that there are no duplicates in `self.qubits`.

Reply:
Both of these suggestions were very helpful, thanks @jakelishman and @TharrmashasthaPV! I believe I have the functionality as intended now, and I will work on some more tests.